libs/libcommon/cmake/find_jemalloc.cmake

```diff
@@ -1,13 +1,19 @@
-option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ON)
-option (USE_INTERNAL_JEMALLOC_LIBRARY "Set to FALSE to use system jemalloc library instead of bundled" ${NOT_UNBUNDLED})
+# bundled jemalloc only for linux x64
+if (NOT ARCH_FREEBSD AND NOT APPLE)
+    option (ENABLE_JEMALLOC "Set to TRUE to use jemalloc" ON)
+    option (USE_INTERNAL_JEMALLOC_LIBRARY "Set to FALSE to use system jemalloc library instead of bundled" ${NOT_UNBUNDLED})
+endif ()
 
 if (ENABLE_JEMALLOC)
-    if (USE_INTERNAL_JEMALLOC_LIBRARY)
-        set (JEMALLOC_LIBRARIES "jemalloc")
-    else ()
+    if (NOT USE_INTERNAL_JEMALLOC_LIBRARY)
```

Comment:
Why these few lines of code was rearranged?

Reply:
Because old not working on freebsd and apple where we cant use USE_INTERNAL_JEMALLOC_LIBRARY by default.
First we should try to find external lib if unbundled mode.
Then we will fallback to contrib version

Reply:
Correct order should be the following:

1. Define `USE_INTERNAL_JEMALLOC_LIBRARY` build option. On systems (like FreeBSD) where it is not preferred mode, the option should be disabled by default. Otherwise it should be enabled by default.

2. If `USE_INTERNAL_JEMALLOC_LIBRARY` is enabled, then use contrib library. If there is no contrib library, throw fatal error with clear message for user - the build should never continue with unintended options.

3. Consider external libraries from the system only if `USE_INTERNAL_JEMALLOC_LIBRARY` is disabled.